src/VirtualClient/VirtualClient.Core/SystemManagement.cs

```diff
@@ -125,6 +125,20 @@ public async Task<CpuInfo> GetCpuInfoAsync(CancellationToken cancellationToken)
             return info;
         }
 
+        /// <summary>
+        /// Get the logged In Username i.e, username of the user who invoked a command with elevated privileges using the "sudo" command in Unix operating system.
+        /// </summary>
+        public string GetLoggedInUserName()
+        {
+            string loggedInUserName = Environment.GetEnvironmentVariable("SUDO_USER");
+            if (string.IsNullOrEmpty(loggedInUserName))
+            {
+                loggedInUserName = Environment.UserName;
```

Comment:
Double-check for me on this but it was my understanding that Yang had validated that "Environment.UserName" by itself works. I remember there might be a minor distinction when we are launching VC on a VM from the Juno Guest agent which runs as a service/daemon. 

Can you clarify the distinction here?

Reply:
When we manually run VC on VM and run using sudo, Environment.UserName returns "root" on linux. We don't get the actual username who is logged in but we get "root". There are certain cases where we need to use the actual username. e.g: azureuser.

Reply:
When I tried I remember Environemnt.Username worked. Maybe this is distro-version specific

Can we try this? Read Environment.Username first, if it's root, then we read the viarable, and return the one that doesn't say root

Reply:
Let's take this offline. I believe this is only in Juno, where root is the true invoking user.